scipy/stats/_continuous_distns.py

```diff
@@ -7526,10 +7525,10 @@ def interval_contains_root(lbrack, rbrack):
 
         # Shape is free
         fit_shape_lt1 = fit_loc_scale_w_shape_lt_1()
-        ll_lt1 = penalized_nllf(fit_shape_lt1, data)
+        ll_lt1 = self.nnlf(fit_shape_lt1, data)
 
         fit_shape_gt1 = fit_loc_scale_w_shape_gt_1()
-        ll_gt1 = penalized_nllf(fit_shape_gt1, data)
+        ll_gt1 = self.nnlf(fit_shape_gt1, data)
```

Comment:
We shouldn't need to use `penalized_nllf` here. It's an ad hoc modification of the negative log-likelihood function that keeps the log-likelihood finite when the parameters are out of bounds.  If the log-likelihood is infinite, so be it; the comparison below should still be valid.

Reply:
`penalized_nllf` was added generically to lead the optimizer back to the support of parameters when parameters (or starting paramaters) are out of bounds.
I don't know how it affects fit in this distribution, or whether it's still relevant here.
There might be more convergence failures without penalization when out of parameter space.